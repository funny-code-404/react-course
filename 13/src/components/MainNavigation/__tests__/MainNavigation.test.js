import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { MainNavigation } from "..";

configure({ adapter: new Adapter() });

describe('MAIN NAVIGATION', () => {
    const mockFn = jest.fn();

    const props = {
        getDataAction: mockFn,
        title: 'title',
        path: '/path'
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<MainNavigation {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('Component is defined', () => {
        expect(wrapper).toHaveLength(1)
    });
    it('Component should return correct action',() => {
        expect(wrapper.find('li')).toHaveLength(1)
    });
    it(' should call onClick when clicking link',() => {
        wrapper.find('Link').simulate('click')
        expect(mockFn).toHaveBeenCalledTimes(1)
    });
});

// npm run test src/components/MainNavigation