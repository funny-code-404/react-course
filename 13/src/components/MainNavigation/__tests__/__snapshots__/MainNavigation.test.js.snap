// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MAIN NAVIGATION to Match Snapshot 1`] = `
<li>
  <Link
    onClick={[Function]}
    to="/path"
  >
    title
  </Link>
</li>
`;
